assets/src/data/eventespresso/core/reducers.js

```diff
@@ -1,161 +1,106 @@
-			function( entity ) {
-				return String( getEntityPrimaryKeyValues( modelName, entity ) );
-			},
-		);
-		return {
-			...state,
-			dirty: {
-				...state.dirty,
-				[ modelName ]: [
-					...difference( state.dirty[ modelName ], entityIds ),
-				],
-			},
-		};
-	}
-	return state;
-}
-
-/**
- * This reducer sets the dirty property to false for the entity record in the
- * state matching the provided entityId in the action object.
- *
- * @param { Object } state
- * @param { Object } action
- * @return { Object }  The new state if the entity record is flushed and the
- *                       original state if not.
- */
-export function cleanEntityById( state = DEFAULT_CORE_STATE, action ) {
-	const { type, modelName, entityId } = action;
-	if ( type === 'CLEAN_ENTITY' &&
-		state.dirty.hasOwnProperty( modelName ) &&
-		state.entities.hasOwnProperty( modelName ) &&
-		state.entities[ modelName ].hasOwnProperty( entityId )
+export default function receiveEntityRecords( state = DEFAULT_CORE_STATE, action ) {
+	const { type, modelName, entities: incomingEntities = new Map() } = action;
+	if (
+		modelName &&
+		state.entities[ modelName ] &&
+		isMap( incomingEntities ) &&
+		! isEmpty( incomingEntities )
 	) {
-		return {
-			...state,
-			dirty: {
-				...state.dirty,
-				[ modelName ]: [
-					...without( state.dirty[ modelName ], String( entityId ) ),
-				],
-			}
-			,
-		};
-	}
-	return state;
-}
-
-/**
- * Receives entities and adds them to or updates them in the state.
- *
- * Any new entity entities are simply added.  Any entities matching existing
- * entities in the state are updated and if any properties of that entity differ
- * from what's already in the state the record is marked dirty.
- *
- * @param { Object } state
- * @param { Object } action
- * @return {*}  Returns original state if no additions or updates are done.
- *                Returns new state if additions or updates are done.
- */
-export function receiveEntityRecords( state = DEFAULT_CORE_STATE, action ) {
-	const { type, modelName, entities: incomingEntities = [] } = action;
-	if ( type === 'RECEIVE_ENTITY_RECORDS' &&
-		state.entities.hasOwnProperty( modelName ) ) {
-		const entities = keyEntitiesByPrimaryKeyValue( modelName,
-			incomingEntities,
-		);
-		const dirty = getDirtyEntityIds( modelName, state, incomingEntities );
-		return {
-			...state,
-			entities: {
-				...state.entities,
-				[ modelName ]: {
-					...state.entities[ modelName ],
-					...entities,
+		let	updateState = false,
+			entities;
+		switch ( type ) {
+			case 'RECEIVE_ENTITY_RECORDS':
+				// if all incoming keys exist in state already then we don't do
+				// anything
+				if ( isEmpty( difference(
+					Array.from( incomingEntities.keys() ),
+					keys( state[ modelName ] )
+				) ) ) {
+					break;
+				}
+				// replace any incoming entities with existing entities already in the
+				// store so this registry acts as the "authority" for the latest entity.
```

Comment:
> Wouldn't the "latest" entity be the incoming one though?

I could see how my choice of words here could be confusing.  I'll change the wording from:

> ..."authority" for he latest entity 

to

> ..."authority" for the entities.

> Maybe the source of the incoming entities should be noted (if possible) to clarify why we are favouring the existing.

I don't think we want to do that here because the action and reducer is agnostic to source for entities.  It's the _behaviour_ that's important.  Client code needs to know that if they dispatch `RECEIVE_ENTITY_RECORDS` action to the `eventespresso/core` store, that the behaviour is any existing entities in the store for incoming entities with a matching id will be kept and incoming discarded.  So source, is irrelevant.

> As far as I know, JS always passes objects by reference, so any client code modifications to an entity in the store will immediately be reflected in any other client code that has retrieved that same entity.
Therefore it is never necessary for client code to have to manually update the entities within the store, correct?

This is incidentally why currently the `eventespresso/lists` store resolver has any entity values retreived from a request _replaced_ with any existing entities in the `eventespresso/core` store.  That way if any updates to entities are made they are immediately reflected anywhere that entity is used in lists.  That's the theory anyways.

Reply:
oops ya forgot.